dotCMS/src/main/java/com/dotcms/security/apps/AppsAPI.java

```diff
@@ -201,6 +221,19 @@ void removeSecretsForSite(Host host, User user)
     void resetSecrets(User user)
                     throws DotDataException, IOException;
 
+    /**
+     * Validate the incoming params and match them with the params described by the respective appDescriptor yml.
+     * This method takes a Map of Optional<char[]> As this is a middle ground object representation
+     * that can be mapped from a saved  AppSecrets or an incoming SecretForm
+     * if the param isn't included in the map it means it wasn't sent..
+     * if the param was sent empty that would be represented as an empty optional.
+     * I'm using optional since null vales on map triggers warnings
+     * @param params
+     * @param appDescriptor
+     */
+    void validateForSave(final Map<String, Optional<char[]>> params,
```

Comment:
use Optional like method arguments is not a goos practice

https://stackoverflow.com/questions/31922866/why-should-java-8s-optional-not-be-used-in-arguments

Reply:
This is  NOT an optional argument. It a map with optional values and right there in the java doc is the reason why I choose to do that. Beside I once pointed the same case in our code base and everyone decided they were fine with such thing.